docs/voyageai.md

```diff
@@ -0,0 +1,182 @@
+- [Handle API keys using pgai from python](#handle-api-keys-using-pgai-from-python)
+
+### Handle API keys using pgai from psql
+
+The api key is an [optional parameter to pgai functions](https://www.postgresql.org/docs/current/sql-syntax-calling-funcs.html).
+You can either:
+
+* [Run AI queries by passing your API key implicitly as a session parameter](#run-ai-queries-by-passing-your-api-key-implicitly-as-a-session-parameter)
+* [Run AI queries by passing your API key explicitly as a function argument](#run-ai-queries-by-passing-your-api-key-explicitly-as-a-function-argument)
+
+#### Run AI queries by passing your API key implicitly as a session parameter
+
+To use a [session level parameter when connecting to your database with psql](https://www.postgresql.org/docs/current/config-setting.html#CONFIG-SETTING-SHELL)
+to run your AI queries:
+
+1. Set your Voyage AI key as an environment variable in your shell:
+    ```bash
+    export VOYAGE_API_KEY="this-is-my-super-secret-api-key-dont-tell"
+    ```
+1. Use the session level parameter when you connect to your database:
+
+    ```bash
+    PGOPTIONS="-c ai.voyage_api_key=$VOYAGE_API_KEY" psql -d "postgres://<username>:<password>@<host>:<port>/<database-name>"
+    ```
+
+1. Run your AI query:
+
+   `ai.voyage_api_key` is set for the duration of your psql session, you do not need to specify it for pgai functions.
+
+    ```sql
+    SELECT * FROM ai.voyageai_embed('voyage-3-lite', 'sample text to embed');
+    ```
+
+#### Run AI queries by passing your API key explicitly as a function argument
+
+1. Set your Voyage AI key as an environment variable in your shell:
+    ```bash
+    export VOYAGE_API_KEY="this-is-my-super-secret-api-key-dont-tell"
+    ```
+
+2. Connect to your database and set your api key as a [psql variable](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-VARIABLES):
+
+      ```bash
+      psql -d "postgres://<username>:<password>@<host>:<port>/<database-name>" -v voyage_api_key=$VOYAGE_API_KEY
+      ```
+   Your API key is now available as a psql variable named `voyage_api_key` in your psql session.
+
+   You can also log into the database, then set `voyage_api_key` using the `\getenv` [metacommand](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-META-COMMAND-GETENV):
+
+      ```sql
+      \getenv voyage_api_key VOYAGE_API_KEY
+      ```
+
+3. Pass your API key to your parameterized query:
+    ```sql
+    SELECT *
+    FROM ai.voyageai_embed('voyage-3-lite', 'sample text to embed', api_key=>$1)
+    ORDER BY created DESC
+    \bind :voyage_api_key
+    \g
+    ```
+
+   Use [\bind](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-META-COMMAND-BIND) to pass the value of `voyage_api_key` to the parameterized query.
+
+   The `\bind` metacommand is available in psql version 16+.
+
+4. Once you have used `\getenv` to load the environment variable to a psql variable
```

Comment:
This does not follow. You talked about connecting and setting your key as a psql variable. Using getenv was an option you added in that step. So, this should be related to the getenv process.